ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-sort/result-sort.tsx

```diff
@@ -0,0 +1,137 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+import styled from '../../styles/styled-components'
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import { WithBackboneProps } from '../../container/backbone-container'
+import { Button, buttonTypeEnum } from '../button'
+
+const SortItemCollectionView = require('../../../component/sort/sort.view.js')
+const Backbone = require('backbone')
+const user = require('../../../component/singletons/user-instance.js')
+
+type Props = {
+  collection: Backbone.Collection<Backbone.Model>
+} & WithBackboneProps
+
+type State = {
+  collection: Backbone.Collection<Backbone.Model>
+}
+
+const Root = styled.div`
+  display: block;
+  padding: 10px;
+  min-width: 500px;
+
+  .editor-footer {
+    padding-top: 10px;
+  }
+
+  .footer-remove {
+    display: inline-block;
+    width: 100%;
+  }
+
+  .footer-remove-has-sort {
+    display: inline-block;
+    width: 50%;
+  }
+`
+
+export default hot(module)(
+    class ResultSort extends React.Component<Props, State> {
+      constructor(props: Props) {
+        super(props)
+        var resultSort = user
+          .get('user')
+          .get('preferences')
+          .get('resultSort')
+
+        this.state = {
+          collection: new Backbone.Collection(resultSort),
+        }
+      }
+
+      handleSort = () => {
+        user
+          .get('user')
+          .get('preferences')
+          .get('resultSort')
+      }
+
+      removeSort = () => {
+        user
+          .get('user')
+          .get('preferences')
+          .set('resultSort', undefined)
+        user
+          .get('user')
+          .get('preferences')
+          .savePreferences()
+      }
+
+      saveSort = () => {
+        var sorting = this.state.collection.toJSON()
+        user
+          .get('user')
+          .get('preferences')
+          .set('resultSort', sorting.length === 0 ? undefined : sorting)
+        user
+          .get('user')
+          .get('preferences')
+          .savePreferences()
+      }
+
+      render() {
+        const hasSort =
+          this.state.collection && this.state.collection.length > 0
+        return (
+          <Root>
+            <MarionetteRegionContainer
+              view={SortItemCollectionView}
+              viewOptions={() => ({
+                collection: this.state.collection,
+              })}
+            />
+
+            {hasSort ? (
+              <div className="editor-footer">
+                <Button
+                  className="footer-remove-has-sort"
+                  buttonType={buttonTypeEnum.negative}
+                  text="Remove Sort"
+                  onClick={this.removeSort}
+                />
+                <Button
+                  buttonType={buttonTypeEnum.positive}
+                  className="footer-remove-has-sort"
```

Comment:
I think it would be good to create a separate styling for this button instead of reusing the `footer-remove-has-sort` styling

Reply:
to what end?  they are intended to have the same styling when there is an existing sort in user preferences.

Reply:
If you wanted to style these buttons differently based on props, say showing or hiding buttons, then separating these classes would be useful. It looks like you could keep both buttons without the ternary, but `hasSort` to the Root styled div and show/hide and resize the buttons based on that prop